include/eve/module/core/regular/impl/convert.hpp

```diff
@@ -139,4 +126,179 @@ namespace eve::detail
       }
     }
   }
+
+  //================================================================================================
+  // wide<->wide convert
+
+  //================================================================================================
+  // wide<->wide default convert implementation
+  template<typename In, typename Out>
+  EVE_FORCEINLINE auto convert_impl(EVE_SUPPORTS(cpu_), In const &v0, as<Out> const &tgt) noexcept
+  {
+    using out_t = as_wide_t<Out, cardinal_t<In>>;
+
+    if constexpr(has_aggregated_abi_v<In>)
+    {
+      // If input is aggregated, we can slice and combine without lose of performance
+      return out_t{eve::convert(v0.slice(lower_),tgt),eve::convert(v0.slice(upper_),tgt)};
+    }
+    else
+    {
+      return map(convert,v0,tgt);
+    }
+  }
+
+  template<simd_value In, scalar_value Out>
+  requires( !product_type<In> && !product_type<Out>)
+  EVE_FORCEINLINE auto convert_(EVE_SUPPORTS(cpu_), In const &v0, as<Out> const &tgt) noexcept
+  {
+    using in_t  = element_type_t<In>;
+    using out_t = element_type_t<Out>;
+
+    // Converting T to T is identity
+    if constexpr(std::same_as<in_t, Out>)
+    {
+      return v0;
+    }
+    else
+    {
+      // EVE treats out of range conversion to be errors and asserts over them
+      if constexpr( sizeof(in_t) > sizeof(out_t) )
+      {
+        EVE_ASSERT( eve::all( v0 <= valmax(as<out_t>()) )
+                  , "[EVE] - eve::convert(" << v0
+                                            << ", as<" << rbr::detail::type_name<out_t>::value()
+                                            << ">()) will overflow."
+                  );
```

Comment:
Wait wat, rly? Since when?
I mean, why aren't we following C++ and just trim stuff.

Like, I would rly like to be able to convert `-1` to unsigned.

Reply:
If we want to follow C++, this means all packs/packus code can't be used. So there is a choice to make there.